.github/workflows/publish.yml

```diff
@@ -18,70 +18,103 @@ env:
+            if [[ ! -z "${TAG_OVERRIDE}" ]]
+            then
+              echo ::set-output name=TAG::${TAG_OVERRIDE}
+            else
+              echo ::set-output name=TAG::$(git describe --tags)
+            fi
+
+        - name: Construct dockerhub and GCR image names
+          id: image-name
+          run: |
+            echo ::set-output name=DOCKERHUB_NAME::${DOCKERHUB_ORG}/${SERVICE_NAME}:${{ steps.tag.outputs.TAG }}
+            echo ::set-output name=GCR_NAME::us.gcr.io/${GOOGLE_PROJECT}/${SERVICE_NAME}:${{ steps.tag.outputs.TAG }}
+
+        - name: Build image locally with jib
+          run: |
+            ./gradlew --build-cache :service:jibDockerBuild \
+            --image=${{ steps.image-name.outputs.DOCKERHUB_NAME }} \
+            -Djib.console=plain
+
+        - name: Login to Docker Hub
+          uses: docker/login-action@v1
+          with:
+            username: dsdejenkins
+            password: ${{ secrets.DSDEJENKINS_DOCKERHUB_PASSWORD }}
+
+        - name: Push dockerhub image
+          run: docker push ${{ steps.image-name.outputs.DOCKERHUB_NAME }}
+
+        - name: Re-tag image for GCR
+          run: docker tag ${{ steps.image-name.outputs.DOCKERHUB_NAME }} ${{ steps.image-name.outputs.GCR_NAME }}
+
+        - name: Set up gcloud
+          uses: google-github-actions/setup-gcloud@v0
+
+        - name: Authenticate to Google Cloud
+          uses: 'google-github-actions/auth@v0'
+          with:
+            # Centralized in dsp-tools-k8s; ask in #dsp-devops-champions for help troubleshooting
+            workload_identity_provider: 'projects/1038484894585/locations/global/workloadIdentityPools/github-wi-pool/providers/github-wi-provider'
+            service_account: 'gcr-publish@broad-dsp-gcr-public.iam.gserviceaccount.com'
+
+        - name: Explicitly auth Docker for GCR
+          run: gcloud auth configure-docker --quiet
+
+        - name: Push GCR image
+          run: docker push ${{ steps.image-name.outputs.GCR_NAME }}
+
+  - update-helm-chart-job:
+      if: startsWith(github.ref, 'refs/tags/') || github.event.inputs.tagOverride != ''
+      runs-on: ubuntu-latest
+      permissions:
+        contents: 'read'
+        id-token: 'write'
+
+      steps:
+        - name: Clone Cromwhelm
+          uses: actions/checkout@v2
+          with:
+            repository: broadinstitute/cromwhelm
+            token: ${{ secrets.BROADBOT_GITHUB_TOKEN }} # Has to be set at checkout AND later when pushing to work
+            path: cromwhelm
+
+        - name: Update CBAS Helm Chart in cromwhelm
+          env:
+            BROADBOT_GITHUB_TOKEN: ${{ secrets.BROADBOT_GITHUB_TOKEN }}
+          run: |
+            set -e
+            cd cromwhelm
+            git checkout main
+            ls -la
+            HELM_CUR_TAG=$(grep "/composite-batch-analysis-service:" cromwell-helm/templates/cbas-deployment.yaml | sed "s,.*/composite-batch-analysis-service:,,")
```

Comment:
This might need to be updated to look for version tag in file `cromwhelm/main/terra-batch-libchart/templates/_cbas-api.tpl` because of [this PR in cromwhelm](https://github.com/broadinstitute/cromwhelm/pull/48/files) that merges cromwell-helm with terra-batch-libchart

Reply:
Thanks for the heads-up. I could make it accommodate both the old and new location, but probably makes sense to just block this on that one merging.

...but really this exposes the hackiness of the whole "bash commands to edit a version number in a text file" approach. It will probably break like this again eventually! I wonder if we can just leave off the version to have it default to "most recent". Not familiar with the syntax, whether that's a thing. I'll ask the team and investigate on my own if nobody knows.

Reply:
Ok, a fruitful discussion with Janet indicates no, that is not a good approach. (It may or may not "work" mechanically speaking but is not a good idea and not how we do things).